File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    let answer = [0, 1000001];
    
    let idx1=0, idx2=0, sum=0;
    for(let i=0; i<sequence.length; i++){
        sum += sequence[idx1++];
        
        if(sum > k){
            while(sum-sequence[idx2] >= k)
                sum -= sequence[idx2++];
        }
        
        if(sum === k){
            while(answer[1]-answer[0] > idx1-1-idx2)
                answer = [idx2, idx1-1];
        }
    }
    
    return answer;
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> sequence, int k) {
    vector<int> answer { 0, 1000001 };
    
    // start, end, sum
    int s=0, e=0, sum=0;
    for(int i=0; i<sequence.size(); i++){
        sum += sequence[s++];
        
        if(sum > k){
            while(sum-sequence[e] >= k)
                sum -= sequence[e++];
        }
        
        if(sum == k){
            if(answer[1]-answer[0] > s-e-1)
                answer = { e, s-1 };
        }
    }
    
    return answer;
}